plugins/processors/t128_rate/t128_rate.go

```diff
@@ -0,0 +1,164 @@
+package rename
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+const sampleConfig = `
+	## If more than this amount of time passes between data points, the
+	## previous value will be considered old and the rate will be recalculated
+	## as if it hadn't been seen before. A zero expiration means never expire.
+	# expiration = "0s"
+
+	## For the fields who's key/value pairs don't match, should the original
+	## field be removed?
+	# remove-original = true
+
+[processors.t128_rate.fields]
+	## Replace fields with their rates, renaming them if indicated
+	# "/rate/metric" = "/total/metric"
+	# "/inline/replace" = "/inline/replace"
+`
+
+type T128Rate struct {
+	Fields         map[string]string `toml:"fields"`
+	Expiration     internal.Duration `toml:"expiration"`
+	RemoveOriginal bool              `toml:"remove-original"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	targetFields map[string]target
+	cache        map[uint64]map[string]observedValue
```

Comment:
I think you answered this once before for me but I just want to confirm - this cache is held by the telegraf instance that the `systemd` service is managing, correct? Meaning its all in-memory and low overhead and this is only a single sample per metric-id, yes?

Reply:
Yes, the telegraf executable holds the cache in memory. It has an entry per metric permutation (combination of ID and unique tags).